vector_double.cpp

```diff
@@ -0,0 +1,207 @@
+#include "vector_double.hpp"
+
+
+template <typename T>
+Vector<T>::Vector()
+{
+    size_v = 0;
+    space  = 0;
+    elem   = nullptr;
+
+}
+
+
+// Copy Constructor
+template<class T>
+Vector<T>::Vector(const Vector<T>& b)
+{
+    space = b.space;
+    size_v = b.size_v;
+    elem = b.elem;
+
+}
+template<class T>
+Vector<T>::Vector(Vector<T>&& obj)
+{
+    space = obj.space;
+    size_v = obj.size_v;
+    elem = obj.elem;
+}
+
+//explicit
+template<class T>
+Vector<T>::Vector(unsigned int)
+{
+    size_v = 0;
+    space  = 0;
+    elem   = nullptr;
+}
+
+
+template<class T>
+Vector<T>::~Vector()
+{
+    delete [] elem;
+}
+
+template<class T>
+int Vector<T>:: size() const
+{
+    return size_v;
+}
+
+template<class T>
+int Vector<T>:: capacity()const
+{
+    return space;
+}
+
+template<typename T>
+void Vector<T>::push_back(const T &v)
+{
+
+    if (size_v >= space)
+    {
+        reserve(space +5);
+        elem [size_v++] = v;
+    }
+
+}
+
+template<class T>
+void Vector<T>::pop_back()
+{
+    if(size_v > 0)
+    {
+        elem[size_v -1];
+    }
+}
+
+
+template<class T>
+void Vector<T>::resize(unsigned int size)
+{
+
+    reserve(size);
+    size_v = size;
+}
+
+
+template<class T>
+void Vector<T>::reserve(unsigned int capacity)
+{
+    unsigned int l_Size;
+
+    if(elem == 0)
+    {
+        size_v = 0;
+        space = 0;
```

Comment:
these two lines don't seem right. if the vector is empty, and we try to reserve space for `capacity` elements, then we set size and space to 0?